tony-core/src/main/java/com/linkedin/tony/HistoryFileUtils.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony;
+
+import java.io.IOException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+
+
+public class HistoryFileUtils {
+  private static final Log LOG = LogFactory.getLog(HistoryFileUtils.class);
+
+  public static TonyJobMetadata createMetadataObj(Configuration yarnConf, String appId, long started, long completed,
```

Comment:
This method should actually be inside `TonyJobMetadata ` class, like `newInstance(a, b, c)` 
Reference: http://www.informit.com/articles/article.aspx?p=1216151

Reply:
So a constructor?

Reply:
Something like this: https://gist.github.com/pdtran3k6/95b2c1f31e0db270cdaa3141b93bf7ee?

Reply:
that works